src/plugins/data/server/search/routes.ts

```diff
@@ -38,8 +41,11 @@ export function registerSearchRoute(router: IRouter): void {
       const { strategy } = request.params;
       const signal = getRequestAbortedSignal(request.events.aborted$);
 
+      const [, , selfStart] = await core.getStartServices();
+      const searchStrategy = selfStart.search.getSearchStrategy(strategy);
+
       try {
-        const response = await context.search!.search(searchRequest, { signal }, strategy);
+        const response = await searchStrategy.search(context, searchRequest, { signal });
```

Comment:
Are there any downsides to passing in a context like this?
I can't think of any, but can you?